src/adapters/class-jw-player.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * WP Video Sync: JW Player Adapter.
+ *
+ * @package wp-video-sync
+ */
+
+namespace Alley\WP\WP_Video_Sync\Adapters;
+
+use Alley\WP\WP_Video_Sync\API\JW_Player_API;
+use Alley\WP\WP_Video_Sync\Interfaces\Adapter;
+use DateTimeImmutable;
+use stdClass;
+
+/**
+ * JW Player Adapter.
+ */
+class JW_Player implements Adapter {
+
+	/**
+	 * The date of the last modification to the last batch of videos.
+	 *
+	 * @var ?DateTimeImmutable
+	 */
+	private ?DateTimeImmutable $last_modified_date;
+
+	/**
+	 * The JW Player API.
+	 *
+	 * @var JW_Player_API
+	 */
+	public JW_Player_API $jw_player_api;
+
+	/**
+	 * The start date if the `last_modified_date` does not exist.
+	 *
+	 * @var string
+	 */
+	public string $origin_modified_date;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param JW_Player_API $api Instance of the JW Player API object.
+	 * @param string        $origin_modified_date The date of the used if the `last_modified_date` does not exist.
+	 */
+	public function __construct( JW_Player_API $api, string $origin_modified_date = '' ) {
+		$this->jw_player_api        = $api;
+		$this->origin_modified_date = ! empty( $origin_modified_date ) ? $origin_modified_date : date( 'Y-m-d' );
```

Comment:
The plugin keeps track of the last modified date via an option, so it shouldn't also need to be kept track of by another property here. If what you want to do is to set a minimum date to override the plugin default of 1/1/1970, then you should update the option with the user-provided value if it isn't already set (see https://github.com/alleyinteractive/wp-video-sync/blob/develop/src/class-sync-manager.php#L97).

The purpose of the `get_last_modified_date()` function is to return the value of the last modified date after a batch of videos has been imported, because where that value comes from is going to be different from adapter to adapter - it shouldn't need to be read until after it's set by the `get_videos` function.

Reply:
So this update was to correct this value from not being initialized because it was typed. It seems like it it is coming from a case where the `wp_video_sync_last_sync` option doesn't exist. I believe what is happening is that when the option does not exist, this https://github.com/alleyinteractive/wp-video-sync/blob/develop/src/class-sync-manager.php#L99 is `not true`, and this method https://github.com/alleyinteractive/wp-video-sync/blob/develop/src/class-sync-manager.php#L116 is throwing the fatal: 

```
Fatal error: Uncaught Error: Typed property Alley\WP\WP_Video_Sync\Adapters\JW_Player_7_For_WP::$last_modified_date must not be accessed before initialization in /var/www/nyp/wp-content/plugins/wp-video-sync/src/adapters/class-jw-player-7-for-wp.php:31
```

I like the idea of setting the date, so I'll look further in to the best solution fit for this

Reply:
Something else is going on here - the `get_option` call uses the second parameter to provide a default, which is a string, so if the option doesn't exist, it will default to the string for the Unix epoch in ISO-8601 format. If that's failing, it likely means that there _is_ an option saved to the options table with that key, and its value is _not_ a string.

For line 116, it will only ever be called _after_ the `get_videos` method is called, which _sets_ that property value, so there should never be a case where it gets to that part of the code and fails unless the `get_videos` method is failing to set that value.